test/factor_nodes/test_poisson.jl

```diff
@@ -0,0 +1,144 @@
+    @test Message(Poisson, l=2.0) == Message{Poisson, Univariate}(ProbabilityDistribution{Univariate, Poisson}(Dict(:l=>2.0))    )
+    @test Message(Poisson) == Message{Poisson, Univariate}(ProbabilityDistribution{Univariate, Poisson}(Dict(:l=>1.0)))
+    @test_throws MethodError Message(Multivariate, Poisson, l=2.0)
+end
+
+@testset "dims" begin
+    @test dims(ProbabilityDistribution(Poisson)) == 1
+end
+
+@testset "slug" begin
+    @test slug(Poisson) == "Poisson"
+end
+
+@testset "vague" begin
+    @test vague(Poisson) == ProbabilityDistribution(Poisson, l=huge)
+end
+
+@testset "isProper" begin
+    @test isProper(ProbabilityDistribution(Poisson, l=1.0))
+    @test !isProper(ProbabilityDistribution(Poisson, l=0.0))
+    @test !isProper(ProbabilityDistribution(Poisson, l=-1.0))
+end
+
+@testset "unsafe mean and variance" begin
+    @test unsafeMean(ProbabilityDistribution(Poisson, l=2.0)) == 2.0
+    @test unsafeVar(ProbabilityDistribution(Poisson, l=2.0)) == 2.0
+end
+
+
+#-------------
+# Update rules
+#-------------
+
+@testset "SPPoissonOutVP" begin
+    @test SPPoissonOutVP <: SumProductRule{Poisson}
+    @test outboundType(SPPoissonOutVP) == Message{Poisson}
+    @test isApplicable(SPPoissonOutVP, [Nothing, Message{PointMass}])
+
+    @test ruleSPPoissonOutVP(nothing, Message(Univariate, PointMass, m=2.0)) == Message(Poisson, l=2.0)
+end
+
+@testset "SPPoissonLPV" begin
+    @test SPPoissonLPV <: SumProductRule{Poisson}
+    @test outboundType(SPPoissonLPV) == Message{Gamma}
+    @test isApplicable(SPPoissonLPV, [Message{PointMass}, Nothing])
+
+    @test ruleSPPoissonLPV(Message(Univariate, PointMass, m=2.0), nothing) == Message(Gamma, a=3.0, b=1.0)
+end
+
+@testset "VBPoissonOut" begin
+    @test VBPoissonOut <: NaiveVariationalRule{Poisson}
+    @test outboundType(VBPoissonOut) == Message{Poisson}
+    @test isApplicable(VBPoissonOut, [Nothing, ProbabilityDistribution])
+
+    @test ruleVBPoissonOut(nothing, ProbabilityDistribution(Gamma, a=2.0, b=3.0)) == Message(Poisson, l=0.5)
+end
+
+@testset "VBPoissonL" begin
+    @test VBPoissonL <: NaiveVariationalRule{Poisson}
+    @test outboundType(VBPoissonL) == Message{Gamma}
+    @test isApplicable(VBPoissonL, [ProbabilityDistribution, Nothing])
+
+    @test ruleVBPoissonL(ProbabilityDistribution(Poisson, l=2.0), nothing) == Message(Gamma, a=3.0, b=1.0)
+    @test ruleVBPoissonL(ProbabilityDistribution(Univariate, PointMass, m=2.0), nothing) == Message(Gamma, a=3.0, b=1.0)
+end
+
+@testset "averageEnergy and differentialEntropy" begin
+    # TODO: Eventually these need to be implemented, but they require additional work for the Poisson distribution
+    # @test isapprox(differentialEntropy(ProbabilityDistribution(Poisson, l=2.0)), averageEnergy(Poisson, ProbabilityDistribution(Poisson, l=2.0), ProbabilityDistribution(Univariate, PointMass, m=2.0)))
+    @test true == false
```

Comment:
Please implement proper tests for the `averageEnergy` and `differentialEntropy` functions